contracts/core/FeeVault.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright 2024, Polymer Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity 0.8.15;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IFeeVault} from "../interfaces/IFeeVault.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract FeeVault is Ownable, ReentrancyGuard, IFeeVault {
+    function depositSendPacketFee(
+        bytes32 channelId,
+        uint64 sequence,
+        uint256[2] calldata gasLimits,
+        uint256[2] calldata gasPrices
+    ) external payable nonReentrant {
+        uint256 fee = gasLimits[0] * gasPrices[0] + gasLimits[1] * gasPrices[1];
+        if ((fee) != msg.value) {
+            revert IncorrectFeeSent(fee, msg.value);
+        }
+        emit SendPacketFeeDeposited(channelId, sequence, gasLimits, gasPrices);
+    }
+
+    function depositOpenChannelFee(address src, string calldata destPortId, string[] calldata connectionHops)
+        external
+        payable
+        nonReentrant
+    {
+        if (msg.value == 0) {
+            revert NoFeeSent();
+        }
+        emit OpenChannelFeeDeposited(src, destPortId, connectionHops);
```

Comment:
> I think we need to emit all the parameters sent to chanopeninit

so you're saying, instead of just the src address and dest portId, we also need version & ordering to for our 
unique key value? 

> shouldn't we also emit the amount of fees paid?

ah yes, added now :/